File: desktop/src/main/java/com/husseinamro/exchange/api/model/Transaction.java
package com.husseinamro.exchange.api.model;

import com.google.gson.annotations.SerializedName;
public class Transaction {
    @SerializedName("usd_amount")
    public Float usdAmount;
    @SerializedName("lbp_amount")
    public Float lbpAmount;
    @SerializedName("usd_to_lbp")
    public Boolean usdToLbp;
    @SerializedName("id")
    public Integer id;
    @SerializedName("added_date")
    public String addedDate;

    public Transaction(Float usdAmount, Float lbpAmount, Boolean usdToLbp)
    {
        this.usdAmount = usdAmount;
        this.lbpAmount = lbpAmount;
        this.usdToLbp = usdToLbp;
    }

    public Float getUsdAmount() {
        return usdAmount;
    }

    public Float getLbpAmount() {
        return lbpAmount;
    }

    public Boolean getUsdToLbp() {
        return usdToLbp;
    }

    public Integer getId() {
        return id;
    }

    public String getAddedDate() { return addedDate; }
}

File: android/app/src/main/java/com/MounirKhalil/currencyexchange/api/model/Transaction.kt
package com.MounirKhalil.currencyexchange.api.model

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

class Transaction() {
    @SerializedName("added_date")
    var addedDate: String? = null

    @SerializedName("id")
    var id: Int? = null

    @SerializedName("usd_amount")
    var usdAmount: Float? = null

    @SerializedName("lbp_amount")
    var lbpAmount: Float? = null

    @SerializedName("usd_to_lbp")
    var usdToLbp: Boolean? = null
}